### DISCUSSION TITLE
mozgcp.net Request / URL Injection

### DISCUSSION THREAD

**Discussion by influential-eliot** [filips123/PWAsForFirefox#595] [General]
Hiya ... I'm pretty sure that it's not the PWA, but I'm wondering if the PWA can potentially be a part of the solution to addressing this.
 
I've noticed that a number of normal URLs (_pick any_) are being converted to run through this origin on the 'mozgcp.net' domain:
```http
https://prod.outgoing.prod.webservices.mozgcp.net
```
My concern is that I've not noted this before ... it seems new ... and it seems to be Firefox related (ie. not malware).

If the URL was `https://pingpong.com/foom` then the URL would change to: `https://prod.outgoing.prod.webservices.mozgcp.net/v1/WHAT_LOOKS_LIKE_SOME_KIND_OF_IDENTIFYING_STRING/https%3A//pingpong.com/foom`

So ... whilst I don't really want to get started on a whole conversation about 'protective redirects' (_like the built in Office URL protection when I'm using my '_Outlook_' PWA via this Extension on Linux_) I am a bit more curious and concerned about this one. Because this appears to be running at browser level ... and whilst it might well be Mozilla '_doing good_' by protecting my links ... ... I didn't **ask** for this.

Oh, and the second reason it's concerning is that the owner of the domain is hiding behind a privacy protection service. (_ironically located down the road from me!_)
 
So, I'm just wondering if this is something that's ever come up, here ... really ... and if there were plans to handle it / address it.

Or ... is it actually something that the PWA is doing actively to protect links? (_I doubt it, but I suppose I have to ask_) :)

---

I've done a fair bit of research via a fair few StartPage searches and cannot seem to find 'owt about this at any official level of detail. So, y'know, I am not starting this discussion with any expectation of response or entitlement to a reply ... just thought I'd bring it up ... Equally ... it's entirely possible that every single time I've noted it that it was a mail (_in the aforementioned Outlook PWA_) that was made by Mozilla ... 

Anyway ... yeah ... curious!

**Comment by filips123** [comment]
I think that normally, these URLs are only for external links comming from Mozilla websites, like the Firefox Addons page. Where are you seeing them?

**Comment by influential-eliot** [comment]
Thanks, yeah, I presumed that, too ... but wasn't sure.
 
I am pretty sure that I have seen these links in emails within the Outlook PWA (running under this infrastructure) ... so ... yeah ... my 'privacy' nyerk immediately bounced! 😅
